### PR TITLE
Implement movement between rooms

### PR DESCRIPTION
**Pull request by ewk**
The player can now move around their environment while we keep track of the game state.

### CODE REVIEW

File: adventure.go [hunk 1]
```diff
@@ -75,11 +74,24 @@ func loadRooms() {
...
+			}
+		}
+	*/
 }
 
 func main() {
 	loadRooms()
 
+	// TODO start room must be initialized with Visited = True
+	curRoom = rooms[0]
```
Comment (BethMyre) [inline adventure.go:93]: We'll need to make sure we're always starting in the attic.  Should we just hard-code that here, or do you have other ideas about how to do that?

File: parser.go [hunk 2]
```diff
@@ -57,6 +57,30 @@ func listInventory() {
...
+func moveToRoom(exit string) {
+	for _, e := range curRoom.Exits {
+		if e == exit { // check that requested exit is valid
+			for i, j := range rooms { // find the room it leads to
+				if j.Name == exit {
+					curRoom = rooms[i] // if found, the exit is the new current room
+
+					if curRoom.Visited == false { // have we been here before?
+						curRoom.Visited = true
+						rooms[i] = curRoom
```
Comment (BethMyre) [inline parser.go:70]: What does this line do?

**Review by BethMyre — COMMENTED** [review]
Looking good! Just two questions before I approve it.

File: adventure.go (hunk 1 above)
Comment (ewk) [reply adventure.go:93]: The start room will need to be initialized here. It will need to hard coded and marked as visited.

File: parser.go (hunk 2 above)
Comment (ewk) [reply parser.go:70]: If we have not visited the current room before, then mark it as visited. 
After updating the current room's Visited field, replace that room in the slice rooms with the updated version.

**Pull request merged**